File: php/sort/Selection.php
<?php
require("AbstractSort.php");
class Selection extends AbstractSort
{
	public static function sort(array $input)
	{
		$length = count($input);
		for ($i = 0; $i < $length; $i++) {
			$min = $i;
			for ($j = $i + 1; $j < $length; $j++) {
				if ($input[$j] < $input[$min])
					$min = $j;
			}

			$tmp = $input[$i];
			$input[$i] = $input[$min];
			$input[$min] = $tmp;
		}
		return $input;
	}
}

Selection::main();

File: python/sort/selection.py
# -*- coding: utf-8 -*-
import sys
from sort import AbstractSort


class SelectionSort(AbstractSort):
    @classmethod
    def sort(cls, origin):
        length = len(origin)
        for i in range(0, length):
            min_index = i
            for j in range(i+1, length):
                if origin[j] < origin[min_index]:
                    min_index = j

            tmp = origin[i]
            origin[i] = origin[min_index]
            origin[min_index] = tmp


if __name__ == "__main__":
    SelectionSort.main()
    sys.exit(0)
